src/playgroundrl_envs/games/poker.py

```diff
@@ -63,124 +73,121 @@ def after_hand_end(self):
         # DETERMINE THE HANDS THAT SHOULD BE SHOWN (ONES THAT WERENT FOLDED
-        # AND STORE THE CARDS 
+        # AND STORE THE CARDS
         for player_id in self.game.in_pot_iter():
             cards = [self._card_to_str(c) for c in self.game.hands[player_id]]
             self.hands[player_id] = cards
-        
 
         # Determine winnings by player. The API for this is pretty janky, so this is what we're
-        # stuck with 
+        # stuck with
         settle_history = self.game.hand_history.settle
         total_winnings = {}
-        for pot_id, (amount, best_rank, winners_list) in settle_history.pot_winners.items():
+        for pot_id, (
+            amount,
+            best_rank,
+            winners_list,
+        ) in settle_history.pot_winners.items():
             winning_per_player = amount / len(winners_list)
             for player_id in winners_list:
-
                 if player_id not in total_winnings:
                     total_winnings[player_id] = 0
                 total_winnings[player_id] += winning_per_player
 
             self.game.pots[pot_id]
-        
-        # Determine reward for each player 
+
+        # Determine reward for each player
         reward = {}
         for player_id, player in self.players.items():
             current_chips = self.game.players[player_id].chips
             chips_spent = self.chips_at_round_start[player_id] - current_chips
             # Initialize reward to chip delta for the round
-            reward[player_id] = - chips_spent
-            self.reward = reward 
+            reward[player_id] = -chips_spent
+            self.reward = reward
 
-        self.last_round_state = {'hands': self.hands, 'winnings': total_winnings}
+        self.last_round_state = {"hands": self.hands, "winnings": total_winnings}
 
         # Start next hand
         self.start_hand()
 
     def _card_to_str(self, card: Card):
         return Card.INT_SUIT_TO_CHAR_SUIT[card.suit] + str(card.rank + 2).zfill(2)
 
-    def submit_action(self, action, player_sid = ""):
+    def submit_action(self, action, player_sid=""):
         player = self.players[self.game.current_player]
 
         if player.sid != player_sid:
-            return False
-        
+            raise PlaygroundInvalidActionException("Not your turn")
 
         action = json.loads(action)
-        type = STR_TO_ACTION_TYPE[action['action_type']]
+        type = STR_TO_ACTION_TYPE[action["action_type"]]
 
-        try: 
+        try:
             if type == ActionType.RAISE:
-                total = action['total']
-                self.game.take_action(type, total = total)
-            else: 
+                total = action["total"]
+                self.game.take_action(type, total=total)
+            else:
                 self.game.take_action(type)
         except ValueError as e:
-            # return False
-            raise e
+            raise PlaygroundInvalidActionException("Raise must be at least {}", total)
```

Comment:
You can use an f string here. Also, I think this will display the current total, not the required total. Something like f"Raise must be at least {self.game.min_raise()}." 

https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python